### ISSUE TITLE
`volatile float` serialized as `int`

### ISSUE THREAD

**Issue by stormtroober** [bblanchon/ArduinoJson#1557]
With the last update of the library, i get floats variable serialized as ints variable.
I'm using a ESP32.

This is the code you need:
```
volatile float distance;
volatile int state;

StaticJsonDocument<128> doc;

```

```
   doc["type"] = "data";
   doc["state"] = state;
   Serial.println(distance);
   doc["distance"] = distance;
   serializeJson(doc, msg);
   doc.clear();
   Serial.println(msg);
```

From the prints i get:
5.74
{"type":"data","state":2,"distance":5}

I **don't** get this issue using a non volatile variable.
Bye

**Comment by bblanchon** [comment]
Hi @aleshark87,

Thank you very much for reporting this issue 👍 
I'll fix the bug in branch `6.x` very soon.

As a workaround, you can cast to `float`:

```c++
doc["distance"] = (float)distance;
```

Best regards,
Benoit

**Comment by bblanchon** [comment]
The fix is included in [ArduinoJson 6.18.1](https://github.com/bblanchon/ArduinoJson/releases/tag/v6.18.1)
